### PR TITLE
Make WANT_UFE_BUILD def public.

### PR DESCRIPTION
**Pull request by HamedSabri-adsk**
Make WANT_UFE_BUILD def public per discussion in issue https://github.com/Autodesk/maya-usd/issues/589

### CODE REVIEW

**Comment by seando-adsk** [comment]
@kxl-adsk Can you review the change I just pushed to fix the failing test.

File: lib/mayaUsd/ufe/StagesSubject.cpp [hunk 1]
```diff
@@ -316,15 +318,21 @@ void StagesSubject::stageChanged(UsdNotice::ObjectsChanged const& notice, UsdSta
...
 void StagesSubject::onStageSet(const MayaUsdProxyStageSetNotice& notice)
 {
-	// We should have no listerners and stage map is dirty.
-	TF_VERIFY(g_StageMap.isDirty());
-	TF_VERIFY(fStageListeners.empty());
-
-	StagesSubject::Ptr me(this);
-	for (auto stage : ProxyShapeHandler::getAllStages())
+	// Handle re-entrant onStageSet
+	if (!stageSetGuardCount.load())
```
Comment (kxl-adsk) [inline lib/mayaUsd/ufe/StagesSubject.cpp:322]: You will have atomic load, but the store on line 324 is not guaranteed to happen without any other thread reading the current value of stageSetGuardCount. Here is how you can make it (and then remove the line 324)
```suggestion
	bool expectedState = false;
	if (stageSetGuardCount.compare_exchange_strong(expectedState,true))
```

**Comment by seando-adsk** [comment]
Thanks @kxl-adsk. That was so cool I committed the change right from the PR (after testing it locally, of course).

**Comment by kxl-adsk** [comment]
Almost perfect...I just forgot to remove the assignment from line 325.

**Comment by seando-adsk** [comment]
@kxl-adsk Done.

**Review by kxl-adsk — APPROVED** [review]

**Review by kxl-adsk — APPROVED** [review]

**Pull request merged**